### DISCUSSION TITLE
[BUG] Partner Sharing - Unable to delete photo

### DISCUSSION THREAD

**Discussion by yurividal** [immich-app/immich#9441] [Feature Request]
### The bug

When partner sharing on the time-line is enabled, the user who doesn't own the picture cannot delete a shared picture. The delete button does nothing.
If this is by design, then the button needs to be hidden, or some sort of UI element to tell the user he can't delete partner's photos.
Even better would be if there was an option in the settings to allow a user to do such a thing.

### The OS that Immich Server is running on

Ubuntu 22.04

### Version of Immich Server

V1.88.1

### Version of Immich Mobile App

V1.88.1

### Platform with the issue

- [X] Server
- [X] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
Same as example one.
```

### Your .env content

```Shell
Same as example one
```

### Reproduction steps

```bash
1. Enable partner sharing
2. Enable shared photos on timeline
3. Try to delete a photo that belongs to your partner. Mobile and Web
...
```

**Comment by BarrettLowe** [comment]
Looks like the following may be the culprit:
https://github.com/immich-app/immich/blob/8a2b36ad55169a6befe4a23bfac8e4f8b697af43/web/src/lib/components/photos-page/actions/delete-assets.svelte#L40

I don't know typescript so I'm having a hard time tracking this backward but at an initial glance, it may work to change this to use getAssets instead...?

**Comment by alextran1502** [comment]
This is as designed. Let me think of a way to make it less confused

**Comment by yurividal** [comment]
> This is as designed. Let me think of a way to make it less confused

Thought it would be as design. Indeed a UI hint is needed here.

Even better would be if there was a option to allow deletion of partners photos.

(Example: my wife is constantly taking pictures of receipts or things she only needs for a few minutes, and never remembers to delete them from immich. I'd love to be able to clean these every now and then, instead of filling our precious storage space with useless pictures)

**Comment by Fl0ete** [comment]
> > This is as designed. Let me think of a way to make it less confused
> 
> Thought it would be as design. Indeed a UI hint is needed here.
> 
> Even better would be if there was a option to allow deletion of partners photos.
> 
> (Example: my wife is constantly taking pictures of receipts or things she only needs for a few minutes, and never remembers to delete them from immich. I'd love to be able to clean these every now and then, instead of filling our precious storage space with useless pictures)

I'd also love to see this. In my case, I am always the one "thinning out" the photos, deleteing blurred ones, keeping only the best shot of many of the same motive. And I would love to just show my partners photos in my timeline (especially with kids), but having them raw is not really satisfying.

Kind of a big ask, but for me the optimal solution would be something like this: I mark the pictures for deletion and my partner gets a notification/hint to look over them and confirm. This way, she doesn't have the work and can still be sure I delete nothing important to her.

**Comment by Jab2870** [comment]
I'm not sure if this is really the correct place to ask, but I would really like to see the partner sharing to allow a lot more control.

Deleting photos is certainly part of that, but other stuff like being able to add a partner's photos to an album etc.

**Comment by andrewdcaudle** [comment]
I concur with a setting that would allow partners to delete the other's images.

**Comment by Tonylolo** [comment]
Yes , very good idea, partner would become a real user. 2 users of 1 photo server, a lot of case for family i think.
Thk for taking in consideration

**Comment by krin-san** [comment]
> I concur with a setting that would allow partners to delete the other's images.

I actually tried to find if this option exists on `/user-settings?isOpen=partner-sharing` page. Imagined, it could look like that:

![Screen Shot 2024-04-19 at 00 01 47](https://github.com/immich-app/immich/assets/2318002/c83d7a2e-9864-435f-a043-ec10d65657f8)

**Reply by flsabourin** [reply]
add a toggle option like "Requires your validation before your partner removes photo"

**Comment by jrasm91** [comment]
This is working as designed and a limitation of the current partner sharing implementation, so I'm going to convert this to a discussion where different enhancement options can be discussed instead.

**Comment by eygraber** [comment]
I think the _other app_ has a somewhat decent model for this. I can "save" photos shared with me, and then that photo becomes mine. My faces are applied to it, I can edit it, and I can delete it. I'm pretty sure they're not duplicating the photo itself, and I just get my own metadata.

I would love to share faces and exif data as well, but I guess that's a separate issue.

**Comment by yurividal** [comment]
Thanks for converting this into a discussion.
I believe the key point here is "choice".
In my opinion, a simple checkbook, or drop-down to allow full control over shared items should be added to the partner sharing setup.
As an admin of my immich instance, (and the only one who really cares about my family memories being clean  and organized 😅) it is frustrating to have to borrow my partners phone in order to delete a bunch of random screenshots and other useless files.

**Reply by ja2ui0** [reply]
I'm in the same boat. To add, getting my partner on board to use immich instead of the other software has been a process, as she doesn't like interfacing with technology or fiddling around with organization. She just wants a backup strategy; I want to leverage that confidence to implement a useful family gallery and a 3-2-1 solution.

She will never add anything to albums or tag items for easy display or organization. That task is mine, as with all the other digital janitor work. My ability to perform that work effectively depends on her ability to logically delegate through configuration what has already been delegated at the highest level of authority :saluting_face:

**Comment by yurividal** [comment]
HI @alextran1502 
Has this been considered by the dev team? Should we get our hopes up that this might one day be implemented?

**Comment by oscarnorberg** [comment]
I am missing this feature as well, in my family I am the one managing our media server and regularly go through our photos, create albums and delete photos and videos that are not needed. The fact that I cannot delete a screenshot or as someone mentioned above, a photo of a receipt is very annoying.

I support having an option under Partner Sharing where you as a user can allow your partner to delete items that you have uploaded.

**Reply by alextran1502** [reply]
We want to have this feature. It requires some extensive refactoring of the permission model which is currently being worked on

**Reply by Digmarx** [reply]
This is good news, as the lack of this feature is a large barrier to use for people with my use case (multiple family members with multiple devices uploading, 1 administrator curating and removing cruft like screenshots, blurry pics, etc)

**Comment by luma-founder** [comment]
+1 to this feature request.  

As a temporary work around, I may create a nightly job that iterates through "Archive Later" and "Trash Later" albums - so we can curate photos together from multiple accounts.

**Comment by yurividal** [comment]
My workaround is simply having a alternative browser (on my phone and on my PC) that is logged in with my wife's account. Then whenever I see junk in immich, I just open that browser to my immich instance and use it to delete the images.

**Reply by anst84** [reply]
If your wife too has Sharing enabled for your library, how do you filter out which photos that is yours? In my case we both have each others photos in our timelines, but I'm unable to find a way to filter on "owner". The only way I can work around this is by disabling sharing.

**Comment by luma-founder** [comment]
Here's **auto-archive** and **auto-trash** scripts for anyone that wants them.   

You can run these on a cron to work around having many users.

https://github.com/benlinton/immich-tools

**Reply by yurividal** [reply]
Thanks. I tried using your approach. The script finds the album, finds the assets and says it marked them as achieved. But, when I open immich they are still there, and not achieved.

I also tried adapting it to use the delete method but couldn't get it to work.
I'm probably doing something wrong...

**Reply by luma-founder** [reply]
I've improved these scripts a lot since posting.  I added them to a git repo for you.  I hope they help.
https://github.com/benlinton/immich-tools

**Reply by luma-founder** [reply]
It probably wasn't obvious until this version - but if you don't include the `--allow-single-album` flag - the archive script will skip assets that have not yet been added to a secondary album (as a safety net to avoid archived photos getting lost in a black hole).  Try using `--verbose` mode to see what's going on under the hood.   Also make sure each user in the config has their own API key or it probably won't work.  Take care.

**Comment by itsgauravjain22** [comment]
+1 for this feature. 

Highly recommended for the photo sharing in family. Partner sharing can have option to allow delete photo.

**Comment by orange-buffalo** [comment]
My workaround is to use this Tampermonkey (or similar) script. It will add a button to the photo view that issues an API call with a token depending on the owner of the asset:
```
// ==UserScript==
// @name         Delete Immich photo
// @version      2025-02-23
// @description  Removes shared photos from Immich
// @author       orange-buffalo
// @match        https://<your host>/*
// @grant        none
// ==/UserScript==

(function() {
    'use strict';

    const usersToKeys = {
        "<id>": "<key>",
    };

    async function fetchAssetOwner(assetId) {
        const response = await fetch(`/api/assets/${assetId}`, {
            method: 'GET',
        });
        if (!response.ok) throw new Error('Failed to fetch asset data');
        const data = await response.json();
        return data.ownerId;
    }

    function extractAssetIdFromUrl() {
        const match = location.pathname.match(/.*?photos\/([a-f0-9\-]+)/);
        if (!match) {
            throw Error("Invalid path, cannot extract asset ID");
        }
        return match[1];
    }

    async function deleteAsset(assetId, apiKey) {
        const response = await fetch('/api/assets', {
            method: 'DELETE',
            headers: {
                'Content-Type': 'application/json',
                'x-api-key': apiKey
            },
            credentials: "omit",
            body: JSON.stringify({ ids: [assetId], force: false })
        });
        if (!response.ok) throw new Error('Failed to delete asset');
        console.log('Deleted Asset ID:', assetId);
    }

    async function deletePhoto() {
        try {
            const assetId = extractAssetIdFromUrl();
            console.log('Extracted assetID: ', assetId);
            const ownerId = await fetchAssetOwner(assetId);
            console.log('Asset owner is: ', ownerId);
            const apiKey = usersToKeys[ownerId];
            if (!apiKey) {
                throw Error('Cannot find key for user: ' + ownerId);
            }
            await deleteAsset(assetId, apiKey);
            alert("Photo moved to Trash");
        } catch (e) {
            alert('Failed to delete: ' + e.message);
        }
    }

    function addButton() {
        const navbar = document.querySelector('[data-testid="asset-viewer-navbar-actions"]');
        if (!navbar || document.querySelector('#custom-log-button')) return;

        const button = document.createElement('button');
        button.id = 'custom-log-button';
        button.textContent = 'Del';
        button.className = 'flex place-content-center place-items-center rounded-full bg-transparent hover:bg-immich-bg/30 text-white p-3 transition-all';
        button.onclick = deletePhoto;

        navbar.appendChild(button);
    }

    function waitForNavbar() {
        const interval = setInterval(addButton, 1000);
    }

    waitForNavbar();
})();
```

For it to work, you need to grab user IDs and [create API keys](https://immich.app/docs/features/command-line-interface/#obtain-the-api-key) for them. Getting user ID might be tricky. One option is to use `/api/users/me` with the corresponding key. Another is to check Immich responses in Network tab and extract `ownerId` from the JSON.

**Comment by arthuragone** [comment]
Deleting photo and editing metadata would be huge improvement of the Partner Sharing setting

**Comment by juanpeich** [comment]
Hello everyone!

Is there any news about being able to delete photos of a partner with whom you're sharing everything?

I manage the photo library at home, and I'd like to be able to delete all the erroneous or incorrect photos that are uploaded, both of me and my wife.

Thank you so much in advance for all your work and dedication.

**Comment by Isusami** [comment]
Same here, I have 2TB of pictures and videos with my partner and I would say we have tons of unnecessary pictures and would be nice if we have access to delete each other's photos.

**Comment by freakyjoe8** [comment]
Same for me. We just came back from holiday and have multiple more or less duplicates as we took pictures of the same scenes. Would be great if the partner could also delete/trash the pictures.

**Comment by bo0tzz** [comment]
Folks, if your comment amounts to "I want this too" then please just use the upvote button on the original post.

**Comment by DomenicOtto** [comment]
Hi, also waiting for this feature. Especially handy when you are both taking pictures and want to avoid doubles. Any idea when this will be implemented?

**Comment by elgatho** [comment]
definetley this is confusing, maybe adding on the main page to turn off partner photos switch, on/off will be a good workaround for now. now i'm checking multiple photos to remove and some of them are removed some not, so we are duplicating job

**Comment by kellergoech** [comment]
Is there any progress on this part?

**Reply by bo0tzz** [reply]
https://justinmayer.com/posts/any-updates/

**Comment by rosbeef** [comment]
Hi 

I think deleting photos without consent is a bit invasive. Maybe if  all partner sharing delete a photo, it really delete it. But if one delete a photo that the other want to preserve, maybe just dereferencing should be done. 

In my mind it should be a big database with all photos owned by immich and deduplicated, then each user have right to see an modify, so duplicate it.

Do I makes me understandable?

**Reply by Novgorod** [reply]
It _is_ invasive, but so is any admin function. Since the admin has full control over all users, it's not some kind of security issue but intended functionality.

It would make administration much easier if _only the admin_ had the same rights over the partner photos as the respective owner (delete, reupload, edit etc.), then I wouldn't have to impersonate a user in order to do maintenance on their partner-shared photos.

**Reply by bo0tzz** [reply]
An admin having implicit rights on other users' assets is never going to happen.

**Reply by Novgorod** [reply]
> An admin having implicit rights on other users' assets is never going to happen.

The admin has by definition (not just) implicit rights over other users' assets because he has full control over the user accounts (e.g. set or change the password) as well as direct access to the actual files and database. That's the sole purpose of an admin account in the first place.

**Comment by flixhsw** [comment]
I am also missing this feature. Actually I would like to be able to do all kind of operations on pictures from my partner: Delete, rotate (once that is implemented), move to archive, etc.
In my opinion a good way to implement that: Add some kind of right management to partner sharing: Just share, allow archive, allow deletion, allow edit metadata (location, timestamp).
Maybe that right management could be done by attaching an API-key with these rights to the share? Just an idea ...

**Comment by Sector14** [comment]
Just adding my support to this feature. 

My use case for this is, we have a few decades worth of family photos that don't really belong in any one users account and it's a lot for any one user to curate and share with everyone via shared albums when only the owner can make changes. A shared user account would accomplish this but the friction this causes for non technical users is high enough that it won't happen.

What I'd like to see would be a way for other users to **request** changes be made to photos they do not own but are shared with them. Requests would be for a photo to be added/removed/moved between shared albums, physically deleted for everyone, tags adding/removing or metadata changes. Perhaps even requesting ownership transfer of a given photo.

This would allow the person sharing the photos to remain in full control but allow the work of curating a huge old archive of photos to be done across multiple people. Think of it like pull request for immich photos :) 

As for the option earlier in the comments to allow delete only to remove it for the one user not all, I don't like this. Delete should remove for everyone if it's shared (by request). Per user option to hide or archive photos only from themselves would be good though with a search option to review those at a later date if you change your mind.

**Comment by KevKo1990** [comment]
I am also supporting this feature. I would need that on a shared album users can be set to manage which pictures are in the album. No need to delete the underlying picture - just removing what is in the album as I shared the album with another person that should be able to manage this album as well.

More specifically, having another right to make the other person more than just editor/Bearbeiter:
<img width="263" height="521" alt="grafik" src="https://github.com/user-attachments/assets/879adb1a-43af-4894-8b4c-107b4dc87d55" />
